File: upp-frontend/src/app/model/dto/ComplaintDto.ts
import { CompliantAssignmentDto } from "./CompliantAssignmentDto";
import { BoardMemberDecisionDto } from "./BoardMemberDecisionDto";

export interface ComplaintDto{
    id: string;
    plagiarisedBook: string;
    originalBook: string;
    boardMemberDecisionDtos: BoardMemberDecisionDto[];
    compliantAssignmentDtos: CompliantAssignmentDto[];
    processInstanceId: string;
    
    jurisdiction: string;
}

File: upp-service/src/main/java/rs/ac/uns/ftn/uppservice/dto/response/ComplaintDto.java
package rs.ac.uns.ftn.uppservice.dto.response;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ComplaintDto {

    private Long id;

    private String plagiarisedBook;
    private String originalBook;
    private Set<BoardMemberDecisionDto> boardMemberDecisionDtos = new HashSet<>();
    private Set<CompliantAssignmentDto> compliantAssignmentDtos = new HashSet<>();
    private String processInstanceId;
    private String jurisdiction;


}
